Project.toml

```diff
@@ -0,0 +1,24 @@
+name = "Vulkan"
+uuid = "9f14b124-c50e-4008-a7d4-969b3a6cd68a"
+authors = ["Cédric Belmant", "Simon Danisch"]
+version = "0.1.0"
+
+[deps]
+DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
+MLStyle = "d8e11817-5142-5d16-987a-aa16d5891078"
+Preferences = "21216c6a-2e73-6563-6e65-726566657250"
```

Comment:
is it necessary to also add these deps to `Vulkan.jl`?

Reply:
DocStringExtensions is for setting up a documentation template, it's not a hard requirement at all but always nice to have. It appears necessary to have it in the project to use it ([AbstractPlotting](https://github.com/JuliaPlots/AbstractPlotting.jl/blob/master/Project.toml) does at least), although this has no influence on the behavior of the library. It can be removed but I don't see any reason not to keep it, since it's very lightweight.

MLStyle is not necessary no, it's used in just one place for pattern matching currently. I can remove it, but it is likely to be handy in the future, and has no extra dependencies at all, so I figured I might keep it for now. If there is a practical reason to limit the dependencies to a minimum, it can be removed, though I would see little benefit there.

Preferences is required to tweak the behavior of the wrapper in an efficient and user-friendly manner with so-called *package options*. It's new to 1.6, and Julia versions < 1.6 just won't be using it and won't have control over the wrapper. An example is whether or not error checking should be performed. It's really easy to tweak this in C/C++ with macros, and that is supposed to be the equivalent in Julia.

Reply:
I thought these deps are only used in VulkanGen, not Vulkan.

Reply:
No, although the major part of Vulkan is generated via VulkanGen, some small things are best wrapped by hand. The user shouldn't know anything about VulkanGen, it should only used by developers to re-generate the wrapper, like Clang for VulkanCore.